tic-tac-toe/src/tic_tac_toe/game.clj

```diff
@@ -13,51 +13,63 @@
 (defn- has-win? [board]
   (board/winning-line? board))
 
-(defn- no-free-spaces? [board]
-  (not (board/free-spaces? board)))
-
 (defn- player-choice []
   (prompt/get-valid-player-option))
 
-(defn- replay? []
-  (= (prompt/get-valid-replay-option) replay/replay-option))
-
-(defn get-players []
-  (players/configure-players (player-choice)))
-
-(defn play-single-move [players board mark]
-  (board/place-mark board mark
-                    ((get players mark) board)))
+(defn no-free-spaces? [board]
+  (not (board/free-spaces? board)))
 
 (defn announce-win [board]
   (writer/display board)
   (writer/win-message (board/winning-symbol board)))
 
-(defn game-over? [board]
-  (or
-    (has-win? board)
-    (no-free-spaces? board)))
-
 (defn announce-draw [board]
   (writer/display board)
   (writer/draw-message))
 
+(defn- game-over? [updated-board]
+  (or (has-win? updated-board)
+      (no-free-spaces? updated-board)))
+
+(defn- announce-result [board]
+ (if (has-win? board)
+   (announce-win board)
+   (announce-draw board)))
+
+(defn- mark-of-next-player [board]
+  (marks/next-mark board))
+
+(defn update-board-with-move [players board mark]
+  (board/place-mark board mark
+                    ((get players mark) board)))
 
 (defn play-move [board players]
-  (let [next-mark (marks/next-mark board)
-        updated-board (play-single-move players board next-mark)]
-    (cond
-      (has-win? updated-board) (announce-win updated-board)
-      (no-free-spaces? updated-board) (announce-draw updated-board)
-      :else
-      (play-move updated-board players))
-
-    (if (game-over? updated-board)
-        (if (replay?)
-            (play-move (empty-board) (get-players))))))
+  (let [next-mark (mark-of-next-player board)
```

Comment:
I did try to move the let block into the loop - so that the loop didn't need to take any parameters - eg:

 `(defn play-move [board players]
   (loop []
   (let [next-mark (mark-of-next-player board)
    updated-board (update-board-with-move players board next-mark)]
    (if (game-over? updated-board)
     (announce-result updated-board)
    (recur)))))`

However I found that the board did not get updated, as though the let block is not executed when it is inside the loop.